main.js

```diff
@@ -0,0 +1,1512 @@
+let selectedTaskId = JSON.parse(
+  localStorage.getItem(LOCAL_STORAGE_SELECTED_TASK_LIST_ID)
+);
+
+const isMobile = screenWidth < 768;
+const setIcon = (element, iconText) => (element.textContent = iconText);
+
+const checkScreenWidth = () => {
+  const taskCardLists = mainPageContainer.querySelectorAll(".task-card-li");
+  isMobile
+    ? taskCardsSetOpenModalByView(taskCardLists)
+    : taskCardsRemoveOpenModalByView(taskCardLists);
+};
+
+function taskCardsSetOpenModalByView(cardLists) {
+  cardLists.forEach((cardList) => {
+    cardList.firstElementChild.setAttribute("data-open-modal-lg", "task-card");
+    cardList.firstElementChild.classList.remove("toggle");
+  });
+}
+
+function taskCardsRemoveOpenModalByView(cardLists) {
+  cardLists.forEach((cardList) => {
+    cardList.firstElementChild.removeAttribute("data-open-modal-lg");
+    cardList.firstElementChild.classList.add("toggle");
+  });
+}
+
+function taskCardToggle(taskId) {
+  if (taskId === undefined) return;
+  const taskCardLi = document.querySelector(`[data-task-id="${taskId}"]`);
+  const taskCardDetailed = taskCardLi.querySelector(".toggle-content");
+  toggleHideShow(taskCardDetailed);
+}
+
+//FORM VALIDATION
+function setNewTaskInputEvents() {
+  const today = new Date().toISOString().substr(0, 10);
+  newTaskPanel.addEventListener("click", newTaskButtonActions, false);
+  newTaskNameInput.addEventListener("input", validateTaskName, false);
+  newTaskDescriptionInput.addEventListener(
+    "change",
+    validateDescription,
+    false
+  );
+  newTaskStartInput.value = today;
+  newTaskStartInput.addEventListener("change", validateTaskStart, false);
+  newTaskDeadlineInput.value = today;
+  newTaskDeadlineInput.addEventListener("change", validateTaskDeadline, false);
+}
+
+function newTaskButtonActions(e) {
+  const buttonType = e.target.dataset.newTaskButton;
+  if (!buttonType) return;
+  buttonType === "create-button" ? getData() : resetForm();
+}
+
+const validateTaskName = function (e) {
+  let element = e.currentTarget;
+  let message;
+  let state;
+  if (newTaskNameInput.value.trim().length < 6) {
+    message = "* Can not be less than 6 characters";
+    state = "error";
+  } else {
+    displayTimeOutValidationStates(element);
+    message = " Looks fine!";
+    state = "valid";
+  }
+  displayValidationStates(element, message, state);
+};
+
+const validateDescription = function () {
+  if (newTaskDescriptionInput.value.trim().length > 0) {
+    console.log("Good to go! ");
+  } else {
+    console.log("This is empty so act like that ");
+  }
+};
+
+const validateTaskDate = function () {
+  const today = new Date().toISOString().substr(0, 10);
+  let startDate = Date.parse(newTaskStartInput.value);
+  let currentDate = Date.parse(today);
+  let finishDate = Date.parse(newTaskDeadlineInput.value);
+  if (startDate > finishDate) {
+    return "Generic Error";
+  } else if (currentDate > finishDate) {
+    return "Spesific Error";
```

Comment:
Typo here please change with
```suggestion
    return "Specific Error";
```